### DISCUSSION TITLE
Problem starting Plausible CE on RHEL 9 - /etc/clickhouse-server is not present

### DISCUSSION THREAD

**Discussion by mattpen** [plausible/analytics#4896] [Self-Hosted Support]
Following the instructions for https://github.com/plausible/community-edition/?tab=readme-ov-file I have installed docker and docker compose from the RHEL repo, I cloned the plausible-ce repo, created a .env and compose.override.yml file.  When I run `docker compose up -d` I get the response "dependency failed to start: container plausible-ce-plausible_events_db-1 is unhealthy".  I ran `docker compose logs plausible_events_db` and observed this error: `Exception: Failed to merge config with '/etc/clickhouse-server/config.d/ipv4-only.xml': Access to file denied: /etc/clickhouse-server/config.d/ipv4-only.xml`. Access to the file seems to be denied because the file does not exist:

```
$ ls -l /etc/clickhouse-server
ls: cannot access '/etc/clickhouse-server': No such file or directory
```

What could be going wrong such that the clickhouse server config files are missing?

**Comment by ruslandoga** [comment]
👋 @mattpen 

Was ls -l /etc/clickhouse-server executed on the host or in the plausible_events_db container? I think the error was different in the logs: access denied instead of missing file.

And re: original error, it might have something to do with RHEL and how it treats Docker bind mounts. However I don't have experience with it. You can try checking permissions on the config files on the host and then in the containers. Maybe adding :ro to mark the mounted files read-only could help.

**Comment by mattpen** [comment]
Hi @ruslandoga - thanks for the insight!

> Was ls -l /etc/clickhouse-server executed on the host or in the plausible_events_db container? I think the error was different in the logs: access denied instead of missing file.

This was on the host. I can't connect to the plausible_events_db container because it is constantly restarting due to the error mentioned. 

> And re: original error, it might have something to do with RHEL and how it treats Docker bind mounts. 

Thanks, I'll investigate down this path

**Reply by ruslandoga** [reply]
Might be relevant:
- https://forums.docker.com/t/bind-mount-not-working/136265 (the suffix is actually Z, not ro)
- https://www.redhat.com/en/blog/container-permission-denied-errors
- https://opensource.com/article/18/2/selinux-labels-container-runtimes?_gl=1*1efgvsk*_gcl_au*NDM2MTE3MzcuMTczNDM3MTM5OA..&extIdCarryOver=true&sc_cid=701f2000001OH6fAAG

So maybe something like this could work:

<kbd>[compose.yml#L16-L25](https://github.com/plausible/community-edition/blob/c29a0f0d134ebc3251c5d5ce76b784fc8631a151/compose.yml#L16-L25)</kbd>

```yml
    volumes:
      - event-data:/var/lib/clickhouse
      - event-logs:/var/log/clickhouse-server
      - ./clickhouse/logs.xml:/etc/clickhouse-server/config.d/logs.xml:Z
      - ./clickhouse/ipv4-only.xml:/etc/clickhouse-server/config.d/ipv4-only.xml:Z
      - ./clickhouse/low-resources.xml:/etc/clickhouse-server/config.d/low-resources.xml:Z
```

**Comment by mattpen** [comment]
Changing `ro` to `Z` didn't seem to have an effect, but I added a+r for plausible-ce/clickhouse and it 's contents and it is now working.  Thanks @ruslandoga for the nedge in the right direction!
